common/nymcoconut/src/scheme/issuance.rs

```diff
@@ -0,0 +1,393 @@
+// Copyright 2021 Nym Technologies SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::TryFrom;
+use std::convert::TryInto;
+
+use bls12_381::{G1Affine, G1Projective, Scalar};
+use group::{Curve, GroupEncoding};
+
+use crate::elgamal::{Ciphertext, EphemeralKey};
+use crate::error::{CoconutError, Result};
+use crate::proofs::ProofCmCs;
+use crate::scheme::setup::Parameters;
+use crate::scheme::BlindedSignature;
+use crate::scheme::SecretKey;
+/// Creates a Coconut Signature under a given secret key on a set of public attributes only.
+#[cfg(test)]
+use crate::Signature;
+use crate::{elgamal, Attribute, ElGamalKeyPair};
+// TODO: possibly completely remove those two functions.
+// They only exist to have a simpler and smaller code snippets to test
+// basic functionalities.
+use crate::traits::{Base58, Bytable};
+use crate::utils::{hash_g1, try_deserialize_g1_projective};
+
+// TODO NAMING: double check this one
+// Lambda
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct BlindSignRequest {
+    // cm
+    commitment: G1Projective,
+    // h
+    pub commitment_hash: G1Projective,
```

Comment:
why is this one field public?

Reply:
In general we need access to this field since it's used for integrity checks at different stages of Coconut scheme. However, to make it look better I replaced it with a function to get this field instead of making it public.